page.go

```diff
@@ -61,6 +99,68 @@ func (p *page) Name() string {
 	return p.name
 }
 
+func (p *page) GetMeta() (Metadata, bool) {
+	if p.metadata.checked > 0 {
+		return p.metadata, p.metadata.checked-1 == 1
+	}
+
+	p.metadata.checked = 1
+	y, err := p.yamlContent()
+	if err != nil || len(y) == 0 {
+		return p.metadata, false
+	}
+	err = yaml.Unmarshal([]byte(y), &p.metadata)
+	if err != nil {
+		return p.metadata, false
+	}
+	p.metadata.checked = 2
+	return p.metadata, true
+}
+
+func (p *page) Title() string {
+	meta, ok := p.GetMeta()
+	if !ok {
+		return p.name
+	}
+	return meta.Title
+}
+
+func (p *page) yamlContent() (string, error) {
+	file, err := os.Open(p.FileName())
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
```

Comment:
Again here we're trying to parse the yaml frontmatter manually, a regex can help a bit here I believe